dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,576 @@
+        _SHAs = yield [SHAs[b]._fit(X, y, **fit_params) for b in _brackets_ids]
+        SHAs = {b: SHA for b, SHA in zip(_brackets_ids, _SHAs)}
+
+        # This for-loop rename estimator IDs and pulls out wall times
+        key = "bracket={}-{}".format
+        for b, SHA in SHAs.items():
+            new_ids = {old: key(b, old) for old in SHA.cv_results_["model_id"]}
+            SHA.cv_results_["model_id"] = np.array(
+                [new_ids[old] for old in SHA.cv_results_["model_id"]]
+            )
+            SHA.model_history_ = {
+                new_ids[old]: v for old, v in SHA.model_history_.items()
+            }
+            for hist in SHA.model_history_.values():
+                for h in hist:
+                    h["model_id"] = new_ids[h["model_id"]]
+                    h["bracket"] = b
+
+        for b, SHA in SHAs.items():
+            n = len(SHA.cv_results_["model_id"])
+            SHA.cv_results_["bracket"] = np.ones(n, dtype=int) * b
+
+        cv_keys = {k for SHA in SHAs.values() for k in SHA.cv_results_.keys()}
+
+        cv_results = {
+            k: [v for b in _brackets_ids for v in SHAs[b].cv_results_[k]]
+            for k in cv_keys
+        }
+        cv_results = {k: np.array(v) for k, v in cv_results.items()}
+
+        scores = {b: SHA.best_score_ for b, SHA in SHAs.items()}
+        best_bracket = max(scores, key=scores.get)
+
+        best_estimator = SHAs[best_bracket].best_estimator_
+
+        estimator_history = {
+            ident: hist
+            for SHA in SHAs.values()
+            for ident, hist in SHA.model_history_.items()
+        }
+
+        # Order history by time
+        history = sum([SHA.history_ for b, SHA in SHAs.items()], [])
+        idx = np.argsort([v["elapsed_wall_time"] for v in history])
+        history = [history[i] for i in idx]
+
+        best_model_id = SHAs[best_bracket].cv_results_["model_id"][
+            SHAs[best_bracket].best_index_
+        ]
+        best_index = np.argwhere(np.array(cv_results["model_id"]) == best_model_id)
+        best_index = best_index.flat[0]
+
+        meta, _ = _get_meta(
+            {b: SHA.history_ for b, SHA in SHAs.items()}, brackets.keys(), SHAs, key
+        )
+
+        self.metadata_ = {
+            "estimators": sum(m["estimators"] for m in meta.values()),
+            "partial_fit_calls": sum(m["partial_fit_calls"] for m in meta.values()),
+            "brackets": meta,
+        }
+
+        self.best_index_ = int(best_index)
+        self.best_estimator_ = best_estimator
+        self.best_score_ = scores[best_bracket]
+        self.best_params_ = cv_results["params"][best_index]
+        self.scorer_ = scorer
+
+        self.model_history_ = estimator_history
+        self.history_ = history
+        self.cv_results_ = cv_results
+
+        self.multimetric_ = SHAs[best_bracket].multimetric_
+        self._SuccessiveHalvings_ = SHAs
+        raise gen.Return(self)
+
+    def metadata(self):
```

Comment:
Two comments about metadata

1. I think this should be a property. The dissonance between this being a method and `.metadata_` being a property is jarring.
2. Can the value for 'brackets` be a `List[Dict]` rather than a `Dict[str, Dict]`. I don't find the key `bracket-=` useful. I'd rather just have a list.

Reply:
It'd also be pleasant if the order of the keys match between `.metadata` and `.metadata_`, but that's minor.

Reply:
> The dissonance between this being a method and .metadata_ being a property is jarring.

:+1: I've refactored to `metadata` and `metadata()`, which is much more clear.